sqlglot/optimizer/unnest_subqueries.py

```diff
@@ -142,9 +146,10 @@ def decorrelate(select, parent_select, external_columns, sequence):
     parent_predicate = select.find_ancestor(exp.Predicate)
 
     # if the value of the subquery is not an agg or a key, we need to collect it into an array
-    # so that it can be grouped
+    # so that it can be grouped. For subquery projections, we use a MAX aggregation instead.
+    agg_func = "MAX" if is_subquery_projection else "ARRAY_AGG"
```

Comment:
can we do real expression building instead of string based?

Reply:
Done